source/com.microsoft.tfs.client.eclipse.ui.egit/src/com/microsoft/tfs/client/eclipse/ui/egit/importwizard/CrossCollectionRepositorySelectControl.java

```diff
@@ -203,26 +208,71 @@ public void run() {
+        // Convert path separators according to the hosting OS rules.
+        return eclipseWorkspaceRoot.toOSString();
 
-        // If the preference is not set then use the home environment variable
-        if (StringHelpers.isNullOrEmpty(workingDirectory)) {
-            workingDirectory = PlatformMiscUtils.getInstance().getEnvironmentVariable(EnvironmentVariables.HOME);
+    }
 
-            // If the home environment variable is not set then use the user
-            // profile (the same logic as eGit)
-            if (StringHelpers.isNullOrEmpty(workingDirectory)) {
-                workingDirectory =
-                    PlatformMiscUtils.getInstance().getEnvironmentVariable(EnvironmentVariables.USER_PROFILE);
+    private String getDefaultRootFolderPreference() {
+        String rootFolderPreference = null;
+
+        // EGit 4.0 and earlier hosted default repository root directory
+        // preference in the UI plugin.
+        final IPreferenceStore uiPrefs = org.eclipse.egit.ui.Activator.getDefault().getPreferenceStore();
+        if (uiPrefs != null) {
+            rootFolderPreference = uiPrefs.getString(DEFAULT_REPOSITORY_DIR_UI_KEY);
+            if (!StringHelpers.isNullOrEmpty(rootFolderPreference)) {
+                return rootFolderPreference;
             }
         }
 
-        return workingDirectory;
+        // Since EGit 4.1 default repository root directory preference is hosted
+        // in the Core plugin.
+        final IEclipsePreferences corePrefs =
+            org.eclipse.egit.core.Activator.getDefault().getRepositoryUtil().getPreferences();
+        if (corePrefs != null) {
+            rootFolderPreference = corePrefs.get(DEFAULT_REPOSITORY_DIR_CORE_KEY, null);
+            if (!StringHelpers.isNullOrEmpty(rootFolderPreference)) {
+                return rootFolderPreference;
+            }
+        }
+
+        // If the preference is not set, then use the home environment variable
+        rootFolderPreference = PlatformMiscUtils.getInstance().getEnvironmentVariable(EnvironmentVariables.HOME);
+        if (!StringHelpers.isNullOrEmpty(rootFolderPreference)) {
+            return rootFolderPreference;
+        }
+
+        // If the home environment variable is not set, then use the user home
+        // directory. (The same logic as in eGit.)
+        rootFolderPreference = LocalPath.combine(FS.DETECTED.userHome().getPath(), "git"); //$NON-NLS-1$
+
+        return rootFolderPreference;
+    }
+
+    private String expandVariables(final String source) {
+        IStringVariableManager variableManager = VariablesPlugin.getDefault().getStringVariableManager();
+
+        try {
+            return variableManager.performStringSubstitution(source);
+        } catch (CoreException e) {
+            log.error("The source string '" + source + "' contains undefined variables", e); //$NON-NLS-1$ //$NON-NLS-2$
+        }
+
+        return null;
```

Comment:
If something goes wrong in expandVariables, I think we should return the string passed in instead of null. By returning null the caller loses information.